### PR TITLE
Enhanced ticket support; plus working queries and tests

### PR DESCRIPTION
**Pull request by d-w-moore**

### CODE REVIEW

**Comment by d-w-moore** [comment]
@trel @korydraughn - Perhaps this should replace #293 due to numerous additions, including queries that seem now free of CAT_SQL_ERR's and another necessary change for basic data access via data_objects.get( ) to work with tickets .

**Comment by d-w-moore** [comment]
Rebased.   Must now:
   - correct codacy issues 
   - add in all columns move all columns from [here](https://github.com/irods/python-irodsclient/blob/888d3b4e6567a9c706eab367badc96b3f6d3bdfc/irods/models.py#L199) into the models namespaced for distinct tables (see first commit above)
   - do final test

**Comment by trel** [comment]
looking good - please resolve any remaining conversations/comments

File: README.rst [hunk 1]
```diff
@@ -670,6 +670,81 @@ As stated, this type of object discovery requires some extra study and effort, b
...
+-------
+
+The irods.ticket.Ticket class lets us issue "tickets" which grant limited
+permissions for other users to access our own data objects (or collections of
+data objects).   As with the iticket client, the access may be either "read"
+or "write".  The recipient of the ticket could be a rodsuser, or even an
+anonymous user.
+
+This is how we would generate a new ticket for access to a logical path - in
+this case, say a collecion containing 1 or more data objects:
```
Comment (trel) [inline README.rst:683]: "collecion" -> "collection"

File: README.rst [hunk 2]
```diff
@@ -670,6 +670,81 @@ As stated, this type of object discovery requires some extra study and effort, b
...
+
+>>> from irods.ticket import Ticket
+>>> new_ticket = Ticket (granting_session)
+>>> ticket_string = new_ticket.issue('read', '/zone/home/my/collection_with_files_for/somebody').string
```
Comment (trel) [inline README.rst:687]: "files" -> "data_objects"

File: irods/test/login_auth_test.py [hunk 3]
```diff
@@ -347,6 +348,43 @@ def test_7(self):
     def test_8(self):
         self.tst0 ( ssl_opt = False, auth_opt = 'pam'    , env_opt = True  )
 
+class TestAnonymousUser(unittest.TestCase):
+
+    def setUp(self):
+#       import pdb; pdb.set_trace();
```
Comment (trel) [inline irods/test/login_auth_test.py:354]: remove before merging

**Review by trel — COMMENTED** [review]
looks good, please get an issue number on the last commit.

**Comment by d-w-moore** [comment]
Yes, and will do final test before #ing.

**Comment by d-w-moore** [comment]
tested, #'d and ready

**Pull request merged**